cfgov/jinja2/v1/_includes/macros/category-slug.html

```diff
@@ -0,0 +1,57 @@
+
+{# ==========================================================================
+
+   category_slug.render()
+
+   ==========================================================================
+
+   Description:
+
+   Render a category slug when given:
+
+   category:           A string.
+
+   path (optional):    If present creates a link with
+                       a path to which the category filter applies.
+                       For example, if the slug is used on a blog article
+                       then path should be '/blog/'.
+                       Remember to leverage vars.path instead of
+                       using the literal string '/blog/'.
+                       Path is used to create the filtered URL:
+                       {{ path }}?filter_category={{ category }}
+
+   classes (optional): Space separated list of class names.
+
+   use_blog_category:  Whether to use the blog category filter or not.
+                       Defaults to false.
+
+   ========================================================================== #}
+
+{% macro render(category, path, classes, use_blog_category=false) %}
+{% import 'macros/category-icon.html' as category_icon %}
+
+{% if path %}
+  {% if use_blog_category %}
+      {% set filter_path = path + '?filter_blog_category=' + category|urlencode|replace('%20', '+') %}
+  {% else %}
+      {% set filter_path = path + '?filter_category=' + category|urlencode|replace('%20', '+') %}
+  {% endif %}
+{% endif %}
+
+{% call _category_link(path, classes) %}
+    {{ category_icon.render(category, 'category-slug_icon') | safe }}
+    <span class="u-visually-hidden">Category:</span>
+    {{ category }}
+{% endcall %}
+{% endmacro %}
+
+{% macro _category_link(href, classes) %}
+{% if href %}
```

Comment:
What do you think about this approach? 

https://gist.github.com/sebworks/5259cec528220e0eb73b